internal/boxcli/shellenv.go

```diff
@@ -24,11 +26,37 @@ type shellEnvCmdFlags struct {
 	runInitHook       bool
 }
 
-func shellEnvCmd() *cobra.Command {
+type shellenvFlagDefaults struct {
+	envForPackageBins bool
+	recomputeEnv      bool
+}
+
+type shellenvFlagDefault func(*shellenvFlagDefaults)
+
+func withEnvForPackageBins(envForPackageBins bool) shellenvFlagDefault {
+	return func(o *shellenvFlagDefaults) {
+		o.envForPackageBins = envForPackageBins
+	}
+}
+
+func withRecompute(recompute bool) shellenvFlagDefault {
+	return func(o *shellenvFlagDefaults) {
+		o.recomputeEnv = recompute
+	}
+}
```

Comment:
I think the `WithOpt` pattern is nice for APIs, but feels a bit heavy handed for an internal function?

Reply:
(That said, this is much better than what it is replacing)

Reply:
Fair point. I was seeking a way to make it optional to override the flag defaults, without forcing each caller to specify it.

The alternative would be the following, which isn't that bad either:
```
func shellEnvCmd(flagDefaults shellEnvFlagDefaults) *cobra.Command {
  ...
}

// devbox commands
shellEnvCmd(&shellEnvFlagDefaults{
     recompute: true,
     envForPackageBins: false,
})
runCmd(&runFlagDefaults{}) // relies on zero-value of envForPackageBins being false
shellCmd(&shellFlagDefaults{}) // relies on zero-value of envForPackageBins being false

// global commands 
shellEnvCmd(&shellEnvFlagDefaults{
     recompute: false,
     envForPackageBins: true,
})
runCmd(&runFlagDefaults{
   envForPackageBins: true,
})
shellCmd(&shellFlagDefaults{
  envForPackageBins: true,
})

```

WDYT?

Reply:
done